plugins/org.python.pydev.core/src/org/python/pydev/core/autoedit/PyAutoIndentStrategy.java

```diff
@@ -830,7 +830,7 @@ public static void customizeParenthesis2(IDocumentCommand command, boolean consi
                         }
                     }
                     if (addRegular) {
-                        if (line.endsWith("__exit__")) {
```

Comment:
You should pre-compile the regexp so that it's not rebuilt every time. 

Also, the second `\\s*` should actually be `\\s+`.

Maybe it'd be nice to have this as an utility on `PySelection`... something as: `PySelection.getFunctionName(line)`: which returns null if it can't find the name or the name of the function if it can be found (then here, you can see if `"__exit__".equals(PySelection.getFunctionName(line))`.

You can use the existing `org.python.pydev.core.docutils.PySelection.FunctionPattern` for that (you may want to create a group for the `\\w*` to get the name of the function later on).

Reply:
i.e.: change the `FunctionPattern` from `"\\s*def\\s+\\w*.*"` to `"\\s*def\\s+(\\w*).*"`